lean/src/test/aime_1994_p3.lean

```diff
@@ -0,0 +1,18 @@
+/-
+Copyright (c) 2021 OpenAI. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kudzo Ahegbebu
+-/
+
+import data.real.basic
+import data.int.modeq
+ 
+theorem aime_1994_p3
+  (x : ℤ  )
+  (f : ℤ  → ℤ )
+  (h0: f(x) + f(x-1) = x^2)
```

Comment:
we want space around binary operation like `-` but not `^`. we also want space around `:`, like what you've done in the other files.
```suggestion
  (h0 : f x + f (x - 1) = x^2)
```